performance/benchmarks/chrono_benchmark_interpolator.cc

```diff
@@ -0,0 +1,391 @@
+/**
+ * file  chrono_benchmark_interpolator.hh
+ *
+ * @author  Alexander Goscinski <alexander.goscinski@epfl.ch>
+ *
+ * @date   22 August 2019
+ *
+ * @brief Benchmarks the interpolator using chrono, main usage for comparisment
+ *        with google benchmarks, but also includes additional benchmarks for
+ *        overhead estimation.
+ *
+ * Copyright  2019 Alexander Goscinski, COSMO (EPFL), LAMMM (EPFL)
+ *
+ * rascal is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation, either version 3, or (at
+ * your option) any later version.
+ *
+ * rascal is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this software; see the file LICENSE. If not, write to the
+ * Free Software Foundation, Inc., 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+
+#include <chrono>
+
+#include "json_io.hh"
+#include "math/interpolator.hh"
+#include "representations/calculator_spherical_expansion.hh"
+
+using namespace rascal;            // NOLINT
+using namespace rascal::math;      // NOLINT
+using namespace rascal::internal;  // NOLINT
+
+/* The number of repetitions to repeat the interpolation method for a number of
+ * iterations. For the initialization process we only use one repititon, because
+ * it takes comparatively long.
+ */
+static constexpr int N_REPETITIONS = 100;
+static unsigned int SEED = 1597463007;
+
+int main() {
```

Comment:
Do we really need to have both the google benchmark version and this one? Could we port the overhead measurement to google bench, or alternatively replace google bench with this?